src/ServiceControl/Hosting/Commands/CustomLivetimes/HostBuilderLifetimeExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Particular.ServiceControl.Commands
+{
+    using global::ServiceControl.Persistence;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Hosting.Internal;
+    using Microsoft.Extensions.Hosting.WindowsServices;
+
+    static class HostBuilderLifetimeExtensions
+    {
+        public static void SetupLifetime(this IHostBuilder hostBuilder, bool runAsWindowsService)
+        {
+            if (runAsWindowsService)
+            {
+                hostBuilder.UseWindowsService();
+                hostBuilder.ConfigureServices(s =>
+                {
+                    s.AddSingleton<WindowsServiceLifetime>();
```

Comment:
do we need this, considering line 15?

Reply:
Yes, as that line only registers an interface implementation and not the implementation separately.